opentelemetry-sdk/src/opentelemetry/sdk/_configuration/__init__.py

```diff
@@ -345,37 +345,54 @@ def _import_id_generator(id_generator_name: str) -> IdGenerator:
     raise RuntimeError(f"{id_generator_name} is not an IdGenerator")
 
 
-def _initialize_components(auto_instrumentation_version):
-    trace_exporters, metric_exporters, log_exporters = _import_exporters(
-        _get_exporter_names("traces"),
-        _get_exporter_names("metrics"),
-        _get_exporter_names("logs"),
+def _initialize_components(**kwargs):
```

Comment:
Overall I think this feature is great for those writing new custom distros.

How would this SDK update affect those existing custom distro+configurators that are doing the manual initialization of components? Example: `_OTelSDKConfigurator._configure` is overridden and does `load_entry_point` with the env vars

Reply:
Could you explain your example more?

The plan is to make this purely additive. Since no distros are currently passing in addition kwargs, there should be no breaking behavior. Env vars will still be used to configure. kwargs will just provide another option on top

Reply:
I think I understand better. What I was thinking of is a custom configurator that subclasses `_OTelSDKConfigurator` but instead of calling `_initialize_components`  as part of `_configure`, its own logic does import and load of sampler and exporter based on env vars. With the kwargs addition this shouldn't change since kwargs are currently ignored by that override anyway.

Would this update introduce an order of precedence where kwargs are preferred over env var?

Reply:
Yeah, params would be preferred over (or merged with) env vars. This follows the approach towards env vars elsewhere in OTel Python. And doing so would be necessary for custom distros to guarantee certain behavior.

In this approach, exporters given by params would be ADDED to those from env vars. Resource attributes would be merged with those from env vars and resource detectors. Since there's only 1 sampler, param would take precedence. Same for logging_enabled